pkg/schedule/coordinator.go

```diff
@@ -74,18 +78,18 @@ type Coordinator struct {
 	cancel context.CancelFunc
 
 	schedulersInitialized bool
-	patrolRegionsDuration time.Duration
 
-	cluster           sche.ClusterInformer
-	prepareChecker    *prepareChecker
-	checkers          *checker.Controller
-	regionScatterer   *scatter.RegionScatterer
-	regionSplitter    *splitter.RegionSplitter
-	schedulers        *schedulers.Controller
-	opController      *operator.Controller
-	hbStreams         *hbstream.HeartbeatStreams
-	pluginInterface   *PluginInterface
-	diagnosticManager *diagnostic.Manager
+	cluster             sche.ClusterInformer
+	prepareChecker      *prepareChecker
+	checkers            *checker.Controller
+	regionScatterer     *scatter.RegionScatterer
+	regionSplitter      *splitter.RegionSplitter
+	schedulers          *schedulers.Controller
+	opController        *operator.Controller
+	hbStreams           *hbstream.HeartbeatStreams
+	pluginInterface     *PluginInterface
+	diagnosticManager   *diagnostic.Manager
+	patrolRegionContext *PatrolRegionContext
```

Comment:
Shall we let the checker controller manage the patrol logic?

Reply:
Move PatrolRegion and patrolRegionContext to checkers
?